app/src/main/java/cc/pogoda/mobile/meteosystem/Main.java

```diff
@@ -125,10 +126,9 @@ public void onCreate() {
 
         favouritiesFile = new FavouritiesFile(fileNames);
 
-        // download all stations from API
-        listOfAllStations = new AllStationsDao().getAllStations();
-
-        Logger.info("[Main][onCreate][listOfAllStations.size() = " + listOfAllStations.size() +  "]");
+        // Download all stations from API in background via JobIntentService. Results are send
+        //back with Broadcast receiver.
+        startGetAllStationsService();
```

Comment:
I don't see how the service looks like, but I have doubts if and how it will cooperate with `recreateListOfFavs` method. You have added a check 
```
        if(listOfAllStations == null) {
            Logger.debug(TAG, "recreateListOfFavs, listOfAllStations=null");
            return;
        }
```
at the beginning of it's (recreateListOfFavs) body, so I assume that in this case it will return immediately, as no station data will be available at this point. And then recreateListOfFavs will be called within a service which is missing in this commit ?